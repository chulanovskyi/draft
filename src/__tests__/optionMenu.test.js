import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import OptionMenu from '../components/todoList/optionMenu/optionMenu.js';

Enzyme.configure({ adapter: new Adapter() });

describe('<OptionMenu />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<OptionMenu applyOptions={jest.fn()} clearOptions={jest.fn()}/>);
  });

  it('should match its snapshot', () => {
    const tree = renderer.create(<OptionMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a `.container__options`', () => {
    expect(wrapper.find('.container__options')).toHaveLength(1);
  });

  it('should not showing options menu after rendering', () => {
    expect(wrapper.find('.options__optionList')).toHaveLength(0);
  });

  it('should show option menu on click', () => {
    wrapper.find('.options__caret').simulate('click');
    expect(wrapper.find('.options__optionList')).toHaveLength(1);
  })

  // it('changes state on editing task name', () => {
  //   wrapper.find('.item__editTask').simulate('click');
  //   const inp = wrapper.find('.item__editInput');
  //   const code = 's'.charCodeAt(0);
  //   inp.simulate('keypress', {which: code, key: 's', keyCode: code}); // Received an action, but input not changing
  //   expect(wrapper.state('editedName')).toBe('read books');
  // });

  // it('show modal on todoItem remove request', () => {
  //   wrapper.find('.item__removeTask').simulate('click');
  //   expect(wrapper.find('.modal__header')).toHaveLength(1);
  // });
  // issue with React 16 Portals (modals) https://github.com/airbnb/enzyme/issues/1150
});
