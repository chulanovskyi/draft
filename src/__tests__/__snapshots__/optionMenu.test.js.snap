// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionMenu /> should match its snapshot 1`] = `
<div
  className="container__options"
>
  <span
    className="fa fa-caret-right options__caret"
    onClick={[Function]}
  >
    Options
  </span>
  <div
    className="hidden"
  >
    <div
      className="optionList__item"
    >
      <span
        className="item__optionLabel"
      >
        Sort
      </span>
      <select
        className="item__sortSelector"
        onChange={[Function]}
      >
        <option
          value="ASC"
        >
          A -&gt; Z
        </option>
        <option
          value="DESC"
        >
          Z -&gt; A
        </option>
      </select>
    </div>
    <div
      className="optionList__item"
    >
      <span
        className="item__optionLabel"
      >
        Filter
      </span>
      <div
        className="item__filterQuery"
      >
        <input
          onChange={[Function]}
          onKeyPress={[Function]}
        />
        <div
          className="filterQuery__matchCase"
        >
          <i
            className="filterQuery__button
                        fa fa-circle-o"
            onClick={[Function]}
          />
          <span>
            match case
          </span>
        </div>
        <div>
          <i
            className="filterQuery__button
                        fa fa-circle-o"
            onClick={[Function]}
          />
          <span>
            exact match
          </span>
        </div>
      </div>
    </div>
    <div
      className="item__wrapApply"
    >
      <div
        className="optionList__item isActiveSelector__item scaleSmaller"
      >
        <span
          className="item__apply"
          onClick={[Function]}
        >
          Clear
        </span>
      </div>
      <div
        className="optionList__item isActiveSelector__item scaleSmaller"
      >
        <span
          className="item__apply"
          onClick={[Function]}
        >
          Apply
        </span>
      </div>
    </div>
  </div>
</div>
`;
